saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh

```diff
@@ -182,7 +182,7 @@ setup_proxy() {
     fi
     PROXY_ENV_FILE=/etc/cdp/proxy.env
     mkdir -p /etc/cdp
-    echo ${PROXY_PROTOCOL}_proxy=$PROXY_URL > $PROXY_ENV_FILE
+    echo https_proxy=$PROXY_URL > $PROXY_ENV_FILE
```

Comment:
what about if the proxy is of type `http_proxy`?

Reply:
yes this seems odd, I would expect to use the correct protocol. maybe we should set both? is there any automated tests for these?

Reply:
The variable name is tied to the target URL and not the proxy' protocol. So, the original code was incorrect.

Earlier, golang used to fall back to `http_proxy` variable in the absence of an `https_proxy` variable.
After a recent golang version update (1.16), the behavior was changed to stop the client falling back to `http_proxy` variable. This broke our NTP proxy support. (See https://go.dev/doc/go1.16#library)

So, if the client needs to connect to `https://example.com`, the variable used would be `https_proxy`, even if the proxy address uses http (eg. `http://proxy.host:5138`)